service/frontend/workflowHandler_test.go

```diff
@@ -119,6 +120,19 @@ func (s *workflowHandlerSuite) SetupTest() {
 	s.mockVisibilityArchiver = &archiver.VisibilityArchiverMock{}
 	s.mockVersionChecker = client.NewMockVersionChecker(s.controller)
 
+	// these tests don't mock the domain handler
+	config := s.newConfig(dc.NewInMemoryClient())
+	s.domainHandler = domain.NewHandler(
+		config.domainConfig,
+		s.mockResource.GetLogger(),
+		s.mockResource.GetDomainManager(),
+		s.mockResource.GetClusterMetadata(),
+		domain.NewDomainReplicator(nil, s.mockResource.GetLogger()),
```

Comment:
nil should be `s.mockProducer` and then you don't need the code change in `TestUpdateDomain_Success_FailOver`

Reply:
I do for a different reason, but good point, thank you